### PR TITLE
Add enum DebugConsoleMode

### PR DESCRIPTION
**Pull request by smart-bo**
#### What it does

Fixes: https://github.com/eclipse-theia/theia/issues/10009

Add enum DebugConsoleMode to **theia.d.ts**

#### How to test

- confirm that references to `DebugConsoleMode` now can correctly `go to definition` in `theia.d.ts`

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/plugin/src/theia.d.ts [hunk 1]
```diff
@@ -9434,6 +9434,10 @@ declare module '@theia/plugin' {
          */
         appendLine(value: string): void;
     }
+    export enum DebugConsoleMode {
+        Separate = 0,
+        MergeWithParent = 1
+    }
```
Comment (msujew) [inline packages/plugin/src/theia.d.ts:9440]: I would like to see additional documentation in here as well. Similar to vscode:

For `Separate`:

```ts
/**
 * Debug session should have a separate debug console.
 */
```

For `MergeWithParent`:

```ts
/**
 * Debug session should share debug console with its parent session.
 * This value has no effect for sessions which do not have a parent session.
 */
```

Also would you mind adding a newline before the enum declaration?

Comment (smart-bo) [reply packages/plugin/src/theia.d.ts]: Thanks for reminding, corrected.

**Review by vince-fugnitto — APPROVED** [review]
The changes look good to me :+1:

**Comment by vince-fugnitto** [comment]
@smart-bo in case you were not aware, you can link issues with [keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) so that they are automatically closed when the pull-request is merged. I've taken the liberty of updating the pull-request description to do so :+1:

**Pull request merged**